src/test/java/gift/service/OrderServiceTest.java

```diff
@@ -0,0 +1,166 @@
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.BDDMockito.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.then;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import gift.product.dto.auth.LoginMember;
+import gift.product.dto.order.OrderDto;
+import gift.product.model.Category;
+import gift.product.model.KakaoToken;
+import gift.product.model.Option;
+import gift.product.model.Order;
+import gift.product.model.Product;
+import gift.product.repository.AuthRepository;
+import gift.product.repository.KakaoTokenRepository;
+import gift.product.repository.OptionRepository;
+import gift.product.repository.OrderRepository;
+import gift.product.repository.WishRepository;
+import gift.product.service.OrderService;
+import java.io.IOException;
+import java.util.Optional;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayNameGeneration;
+import org.junit.jupiter.api.DisplayNameGenerator;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
+@SuppressWarnings("NonAsciiCharacters")
+class OrderServiceTest {
+
+    MockWebServer mockWebServer;
+    ObjectMapper objectMapper;
+
+    @Mock
+    OrderRepository orderRepository;
+
+    @Mock
+    OptionRepository optionRepository;
+
+    @Mock
+    WishRepository wishRepository;
+
+    @Mock
+    AuthRepository authRepository;
+
+    @Mock
+    KakaoTokenRepository kakaoTokenRepository;
+
+    @InjectMocks
+    OrderService orderService;
+
+    @BeforeEach
+    void ObjectMapper_셋팅() {
+        this.objectMapper = new ObjectMapper();
+    }
+
+    @BeforeEach
+    void 가짜_API_서버_구동() throws IOException {
+        mockWebServer = new MockWebServer();
+        mockWebServer.start();
+    }
+
+    @AfterEach
+    void 가짜_API_서버_종료() throws IOException {
+        mockWebServer.shutdown();
+    }
+
+    @Test
+    void 주문_전체_조회() {
+        //given
+        LoginMember loginMember = new LoginMember(1L);
+
+        //when
+        orderService.getOrderAll(loginMember);
+
+        //then
+        then(orderRepository).should().findAllByMemberId(loginMember.id());
+    }
+
+    @Test
+    void 주문_조회() {
+        //given
+        LoginMember loginMember = new LoginMember(1L);
+        Order order = new Order(1L, 1L, 1, "test_message");
+        given(orderRepository.findByIdAndMemberId(1L, 1L)).willReturn(Optional.of(order));
+
+        //when
+        orderService.getOrder(1L, loginMember);
+
+        //then
+        then(orderRepository).should().findByIdAndMemberId(1L, loginMember.id());
+    }
```

Comment:
```suggestion
   @Test
    void 주문_조회() {
        //given
        LoginMember loginMember = new LoginMember(1L);
        Order order = new Order(1L, loginMember.id(), 1, "test_message");
        given(orderRepository.findByIdAndMemberId(1L, loginMember.id())).willReturn(Optional.of(order));

        //when
        orderService.getOrder(order.getId(), loginMember);

        //then
        then(orderRepository).should().findByIdAndMemberId(order.getId(), loginMember.id());
    }
```

그냥 숫자만 맞게 맞춰 넣는 것 보다 요런 식으로 앞에서 만들어놓을 객체의 값을 사용하는게 조금 더 탄탄한 테스트를 만들 수 있습니다.!

지금은 하나의 케이스만 검증하지만 경우에 따라 `@ParameterisedTest`와 `@CSVSource` 등을 사용해 여러 건의 케이스를 동적으로 주입받아 사용하는 경우 이런 식으로 구성할 수 없거든요.!

Reply:
조언 주신 부분에 대해 기존에는 필요성 체감을 못했어서 간과했었는데, 알려주신 @ParameterisedTest와 @CSVSource에 대해 검색해보고 공부해보니 확실히 필요성이 체감이 됐습니다. 감사합니다! 해당 부분 반영하였습니다.

Reply:
필수는 아니지만, 기존에 존재하는 테스트 중 한두개를 다음 단계에서 ParameterisedTest로 바꿔 보는 것도 추천드립니다!